solana/pyth2wormhole/client/src/main.rs

```diff
@@ -301,56 +272,181 @@ async fn handle_attest(
         Duration::from_millis(attestation_cfg.min_rpc_interval_ms),
     ));
 
-    let message_q_mtx = Arc::new(Mutex::new(P2WMessageQueue::new(
-        Duration::from_millis(attestation_cfg.min_msg_reuse_interval_ms),
-        attestation_cfg.max_msg_accounts as usize,
-    )));
+    // The loop below attempts to crank attestations without
+    // interruption. This is achieved by spinning up an up-to-date
+    // symbol set before letting go of the previous one. This Option
+    // preserves the state of potential ongoing scheduling jobs. The
+    // hash is sensitive to both attestation_cfg and on-chain config
+    // changes.
+    let mut old_sched_futs_state: Option<(JoinHandle<_>, u64)> = None; // (old_futs_handle, old_config_hash)
 
-    // Create attestation scheduling routines; see attestation_sched_job() for details
-    let mut attestation_sched_futs = batches.into_iter().map(|(batch_no, batch)| {
-        attestation_sched_job(
-            batch,
-            batch_no,
-            batch_count,
-            n_retries,
-            retry_interval,
-            daemon,
-            rpc_cfg.clone(),
-            p2w_addr,
-            config.clone(),
-            Keypair::from_bytes(&payer.to_bytes()).unwrap(),
-            message_q_mtx.clone(),
-        )
-    });
+    // This loop governs mapping account reloads in daemon mode. It is
+    // broken if daemon != true at the and
+    loop {
+        let start_time = Instant::now();
+        let config = get_config_account(&lock_and_make_rpc(&rpc_cfg).await, &p2w_addr).await?;
+
+        // Use the mapping if specified
+        if let Some(mapping_addr) = attestation_cfg.mapping_addr.as_ref() {
+            // De-escalate crawling errors; A temporary failure to
+            // look up the mapping should not crash the attester
+            match crawl_pyth_mapping(&lock_and_make_rpc(&rpc_cfg).await, mapping_addr).await {
+                Ok(additional_accounts) => {
+                    debug!(
+                        "Crawled mapping {} data:\n{:#?}",
+                        mapping_addr, additional_accounts
+                    );
+                    attestation_cfg.add_symbols(additional_accounts, "mapping".to_owned());
+                }
+                Err(e) => {
+                    error!("Could not crawl mapping {}: {:?}", mapping_addr, e);
+                }
+            }
+        }
+        debug!(
+            "Attestation config (includes mapping accounts):\n{:#?}",
+            attestation_cfg
+        );
 
-    info!("Spinning up attestation sched jobs");
+        info!(
+            "{} symbol groups read, dividing into batches",
+            attestation_cfg.symbol_groups.len(),
+        );
 
-    let results = futures::future::join_all(attestation_sched_futs).await; // May never finish for daemon mode
+        let mut batches: Vec<_> = attestation_cfg
```

Comment:
I think we can compute these only when needed (when config has changed below). Performance-wise it's not much different but it will result in more clear logs.